File: result/metagpt/claude/humaneval/64.c
#include <stdio.h>
#include <string.h>

int vowels_count(const char *s) {
    const char *vowels = "aeiouAEIOU";
    int count = 0;
    int len = strlen(s);

    for (int i = 0; i < len; i++) {
        if (strchr(vowels, s[i]) != NULL) {
            count++;
        }
    }

    if (len > 0 && (s[len - 1] == 'y' || s[len - 1] == 'Y')) {
        count++;
    }

    return count;
}

File: result/metagpt/deepseek/humaneval/64.rs
pub fn vowels_count(s: &str) -> usize {
    let vowels: Vec<char> = "aeiouAEIOU".chars().collect();
    let mut count = s.chars().filter(|c| vowels.contains(c)).count();
    
    if let Some(last_char) = s.chars().last() {
        if last_char == 'y' || last_char == 'Y' {
            count += 1;
        }
    }
    
    count
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_vowels_count() {
        assert_eq!(vowels_count("hello"), 2);
        assert_eq!(vowels_count("why"), 2);
        assert_eq!(vowels_count("aeiou"), 5);
        assert_eq!(vowels_count("AEIOUY"), 6);
        assert_eq!(vowels_count(""), 0);
    }
}